### ISSUE TITLE
Dev Container 0.255.2 : Default remote user as change from root to node

### ISSUE THREAD

**Issue by ijaouani** [microsoft/vscode-remote-release#7307]
- VSCode Version: 1.72.0
- Local OS Version: Windows 11 Pro 22H2
- Remote OS Version: N/A
- Remote Extension/Connection Type: Containers stable v0.255.2 or prerelease 0.256.0

Steps to Reproduce:

1. Create a new devcontainer using for exemple [nodejs template](https://github.com/microsoft/vscode-dev-containers/tree/main/containers/javascript-node) and reopen in container.

<img width="444" alt="image" src="https://user-images.githubusercontent.com/27200712/194539984-99f5fd74-522e-447d-b72f-120706055479.png">

2. In devcontainer.json comment the line 31  :
```
30	// Comment out to connect as root instead. More info: https://aka.ms/vscode-remote/containers/non-root.
31	// "remoteUser": "node"
```
3. After rebuild the user is still node not root :

<img width="529" alt="image" src="https://user-images.githubusercontent.com/27200712/194540471-de9f0b0a-4268-4fb5-b46b-45971dcf4e87.png">

This is working correctly when rolling back to previous stable version 0.251.0.

**Comment by simensol** [comment]
I think this is the reason I suddenly started to get errors when trying to execute `npm install`:

```bash
[2022-10-08T05:59:17.820Z] [05:59:17] Downloaded extension: dbaeumer.vscode-eslint /home/node/.vscode-server/extensionsCache/dbaeumer.vscode-eslint-2.2.6
[2022-10-08T05:59:17.897Z] npm
[2022-10-08T05:59:17.897Z]  
[2022-10-08T05:59:17.897Z] ERR! code
[2022-10-08T05:59:17.897Z]  EACCES
[2022-10-08T05:59:17.898Z] npm ERR!
[2022-10-08T05:59:17.898Z]  syscall
[2022-10-08T05:59:17.898Z]  rmdir
[2022-10-08T05:59:17.899Z] npm 
[2022-10-08T05:59:17.899Z] ERR! 
[2022-10-08T05:59:17.899Z] path
[2022-10-08T05:59:17.899Z]  /workspaces/frontend/node_modules/.bin
[2022-10-08T05:59:17.899Z] 
[2022-10-08T05:59:17.900Z] npm ERR!
[2022-10-08T05:59:17.900Z]  
[2022-10-08T05:59:17.901Z] errno -13
[2022-10-08T05:59:17.901Z] 
[2022-10-08T05:59:17.904Z] npm ERR!
[2022-10-08T05:59:17.904Z]  Error: EACCES: permission denied, rmdir '/workspaces/frontend/node_modules/.bin'
npm 
[2022-10-08T05:59:17.904Z] ERR!
[2022-10-08T05:59:17.904Z]   [Error: EACCES: permission denied, rmdir '/workspaces/frontend/node_modules/.bin'] {
npm
[2022-10-08T05:59:17.904Z]  
[2022-10-08T05:59:17.905Z] ERR!   errno: -13,
[2022-10-08T05:59:17.905Z] npm
[2022-10-08T05:59:17.905Z]  
[2022-10-08T05:59:17.905Z] ERR!
[2022-10-08T05:59:17.905Z]    code: 'EACCES',
npm 
[2022-10-08T05:59:17.906Z] ERR!   syscall: 'rmdir',
[2022-10-08T05:59:17.906Z] npm
[2022-10-08T05:59:17.906Z]  ERR!
[2022-10-08T05:59:17.906Z]    path: '/workspaces/frontend/node_modules/.bin'
[2022-10-08T05:59:17.906Z] npm ERR! }
[2022-10-08T05:59:17.907Z] npm 
[2022-10-08T05:59:17.907Z] ERR! 
npm ERR! The operation was rejected by your operating system.
npm ERR! It is likely you do not have the permissions to access this file as the current user
npm 
[2022-10-08T05:59:17.908Z] ERR! 
[2022-10-08T05:59:17.908Z] 
npm ERR! If you believe this might be a permissions issue, please double-check the
npm ERR! permissions of the file and its containing directories, or try running
[2022-10-08T05:59:17.908Z] npm ERR! the command again as root/Administrator.
[2022-10-08T05:59:17.910Z] 
[2022-10-08T05:59:17.910Z] npm
[2022-10-08T05:59:17.910Z]  
[2022-10-08T05:59:17.911Z] ERR! A complete log of this run can be found in:
npm
[2022-10-08T05:59:17.912Z]  ERR!     /home/node/.npm/_logs/2022-10-08T05_59_16_986Z-debug-0.log
```

**Comment by chrmarti** [comment]
The `remoteUser` is now included in the base image. If you need it to be root, you can specify `"remoteUser": "root"` in the devcontainer.json.

/cc @bamurtaugh @Chuxel

**Comment by simensol** [comment]
`"remoteUser": "root"` fixed my `npm install` issue above for both the `python` and `javascript-node` base images. I had to rebuild the containers without cache to get rid of some other permission errors, though. Maybe mention this explicitly in [the docs](https://code.visualstudio.com/remote/advancedcontainers/add-nonroot-user)?

**Comment by ijaouani** [comment]
> The `remoteUser` is now included in the base image. If you need it to be root, you can specify `"remoteUser": "root"` in the devcontainer.json.
> 
> /cc @bamurtaugh @Chuxel

It's that I did but I think the containers definitions need to be updated to avoid confusion.

**Comment by simensol** [comment]
With `"remoteUser": "root"`, when I attach to a running devcontainer (or do `Developer: Reload Window` from within the devcontainer) the user is no longer `root`, but a user named `vscode` in the devcontainer based on the `python` base image, and a user named `node` in the devcontainer based on the `javascript-node` base image. To get back as `root` user, I have to rebuild the devcontainer(s).

The problem is that I'm [connecting to multiple containers](https://code.visualstudio.com/remote/advancedcontainers/connect-multiple-containers) using one `docker-compose.yml` for both the `python` and `javascript-node` devcontainers. If I start the `python` devcontainer and then try to attach to the `javascript-node` devcontainer, the user is now `node` and not `root`. However, if I rebuild the `javascript-node` devcontainer to get back as `root`, the connection to the `python` devcontainer is lost.

It's the same the other way around: If I start the `javascript-node` devcontainer and then try to attach to the `python` devcontainer, the user is now `vscode` and not `root`. If I rebuild the `python` devcontainer to get back as `root`, the connection to the `javascript-node` devcontainer is lost.

In other words, it is not possible to get into both devcontainers as `root` at the same time.

**Comment by chrmarti** [comment]
@simensol Not sure I fully understand what is going on, but could you try using `"shutdownAction": "none"` in the devcontainer.json files involved? That should prevent the rebuild from affecting the other container.

**Comment by simensol** [comment]
I already have `"shutdownAction": "none"` in both `devcontainer.json` files. However, this (rebuilds affecting the other container) has also been an issue in previous versions, but I have always been able to get back by reloading the window or attaching to an existing devcontainer. Now, that I "loose" the `root` user, this workaround doesn't work anymore.

Rolling back to `0.251.0` fixes the issue where the `"remoteUser": "root"` is not respected when attaching to a running devcontainer (or when I do `Developer: Reload Window` from within the devcontainer). Updating again to `0.255.2` results in the `root` user being available only the first time the devcontainer is created (as [described above](https://github.com/microsoft/vscode-remote-release/issues/7307#issuecomment-1273292811)).

**Comment by mhofman** [comment]
I have a similar but somewhat mirror issue. I have a container I branched off the node-typescript image some time ago. The remote extension just updated and now whatever the setting of `remoteUser`, the container keeps starting as root instead of the `node` user. Any suggestion besides rebuilding the container based on a newer image?

**Comment by chrmarti** [comment]
Added a fix in Dev Containers 0.259.0-pre-release. Could you give that a try and let me know if that fixes it? Thanks.

**Comment by mhofman** [comment]
At the very least for my case, 259 does fix the issue, and a simple `"remoteUser": "node"` works again.

**Comment by simensol** [comment]
`0.259.0-pre-release` seems to have fixed my issues as well. Reloading any of the two devcontainers now resumes with `root` as user every time.

**Comment by chrmarti** [comment]
Fix is now also available in Dev Containers 0.255.3.

**Comment by jhzsquared** [comment]
I'm having a similar issue where "ShutdownAction": "none" is being ignored in my devcontainer.json file-- my container is restarting everytime I exit vscode/the specific devcontainer window.  I've tried the various versions of the devcontainer extension: to include 0.255.3 , 0.259.0 and the newest (0.262.3) with no luck. Not sure why it's not working.

I'm on VSCode version 1.73.
